src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala

```diff
@@ -269,18 +270,38 @@ class SubmissionSpec(_system: ActorSystem) extends TestKit(_system) with FlatSpe
   }
 
   it should "return a successful Submission when given an entity expression that evaluates to a set of entities" in withWorkspaceService { workspaceService =>
-    val submissionRq = SubmissionRequest("dsde", "GoodMethodConfig", "SampleSet", "sset1", Some("this.samples"))
+    val sset = Entity("testset6", "SampleSet",
+      Map( "samples" -> AttributeEntityReferenceList( Seq(
+        AttributeEntityReference("Sample", "sample1"),
+        AttributeEntityReference("Sample", "sample2"),
+        AttributeEntityReference("Sample", "sample3"),
+        AttributeEntityReference("Sample", "sample4"),
+        AttributeEntityReference("Sample", "sample5"),
+        AttributeEntityReference("Sample", "sample6")))))
+
+    runAndWait(entityQuery.save(SlickWorkspaceContext(testData.workspace), sset))
+
+    val submissionRq = SubmissionRequest("dsde", "GoodMethodConfig", sset.entityType, sset.name, Some("this.samples"))
     val rqComplete = Await.result(workspaceService.createSubmission( testData.wsName, submissionRq ), Duration.Inf).asInstanceOf[RequestComplete[(StatusCode, SubmissionReport)]]
 
     val (status, newSubmission) = rqComplete.response
     assertResult(StatusCodes.Created) {
       status
     }
 
-    assert( newSubmission.notstarted.size == 0 )
-    assert( newSubmission.workflows.size == 3 )
+    // the mock exec service always returns 2 successes and a failure and there should be 2 batches
+    assert( newSubmission.notstarted.size == 0 ) // this not started count includes only those with bad inputs, not errors
+    assert( newSubmission.workflows.size == 4 )
+
+    assertResult(Map("sample1" -> "69d1d92f-3895-4a7b-880a-82535e9a096e", "sample2" -> "69d1d92f-3895-4a7b-880a-82535e9a096f", "sample4" -> "69d1d92f-3895-4a7b-880a-82535e9a096e", "sample5" -> "69d1d92f-3895-4a7b-880a-82535e9a096f")) {
+      newSubmission.workflows.map(r => r.entityName -> r.workflowId).toMap
+    }
 
     checkSubmissionStatus(workspaceService, newSubmission.submissionId)
+
+    val submission = runAndWait(submissionQuery.loadSubmission(UUID.fromString(newSubmission.submissionId))).get
+    assert( submission.workflows.size == 4 )
+    assert( submission.notstarted.size == 2 ) // this not starated count includes errors
```

Comment:
I agree that it is confusing here but I don't think the api itself is confusing... anyway it seems like we are going the queuing route so this will all change